File: frontend/types/contributors.ts
export interface Contributor {
  login: string;
  avatar_url: string;
  html_url: string;
  contributions: number;
  percentage: number;
}

export interface ContributorsSummary {
  total_contributors: number;
  contributors: Contributor[];
  bus_factor: number;
  bus_factor_note: string;
}

File: backend/app/schemas/contributors.py
from __future__ import annotations

from pydantic import BaseModel


class Contributor(BaseModel):
    login: str
    avatar_url: str
    html_url: str
    contributions: int
    percentage: float


class ContributorsSummary(BaseModel):
    total_contributors: int
    contributors: list[Contributor]
    bus_factor: int
    bus_factor_note: str
